persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/UserDAOImpl.kt

```diff
@@ -67,6 +67,29 @@ class UserDAOImpl(private val queries: UsersQueries) : UserDAO {
         queries.updateUser(user.name, user.handle, user.email, user.accentId, user.previewAssetId, user.completeAssetId, user.id)
     }
 
+    override suspend fun updateUsers(users: List<UserEntity>) {
+        queries.transaction {
+            users.forEach { user ->
+                queries.updateUser(user.name, user.handle, user.email, user.accentId, user.previewAssetId, user.completeAssetId, user.id)
+                val recordDidNotExist = queries.selectChanges().executeAsOne() == 0L
```

Comment:
Doesn't `selectChanges()` return the number of rows modified in the previous sql statement? what is then `executeAsOne()` statement doing?

Reply:
This `selectChanges()` in effect returns the rows affected by the table operation, the `.executeAsOne()` function allow us to get the result, since the result of a query it's a raw `Query` result, for example when we return the results from a query we usually use `.asFlow()` to get a more manageable type

PS: I added the tests that you suggested =)